### ISSUE TITLE
No optimization, just svg-sprite

### ISSUE THREAD

**Issue by Avcajaraville** [jkphl/svg-sprite#41]
Im struggling trying to make no optimization on the svg. My sources files are fine, but when compressed on a sprite, there is lot of "fails" on the outputted sprite. This have to do with SVO optimization, but Im not able to find out how or where.

Is there any way I can disable svg optimization and just use a compiled sprite from the original svg ?

Thanks a lot !

**Comment by jkphl** [comment]
Hi @Avcajaraville,

you can disable SVG optimization altogether by emptying the default list of `transform`s:

```javascript
var config = {
	transform: []
	/* ... */
}
```

For an example of how to disable certain SVGO plugins (which is most likely what you will want to do to solve your problem) please [see here](https://github.com/jkphl/svg-sprite#b21-pre-defined-transformation-with-custom-configuration-object-values).

Please let me know if this works for you!

Cheers, Joschi

**Comment by Avcajaraville** [comment]
Not really. I disable the svg optimization, and the output its better, but still, some problems happens.

This are some .svg icons I have, they are ok, you can see them properly on the browser, and on image viewer : 

![pic1](https://cloud.githubusercontent.com/assets/2271201/5739381/427a2840-9bf9-11e4-8464-0f0fe9ba68b7.png)

![pic2](https://cloud.githubusercontent.com/assets/2271201/5739387/502d1cfe-9bf9-11e4-90c2-970e00251c55.png)

![pic4](https://cloud.githubusercontent.com/assets/2271201/5739436/d922748c-9bf9-11e4-99c9-79381150fedc.png)

 

And then, when compiled into a sprite (as I said, tried disabling svgo as you explained, and also disabling plugins by hand (all of them) ), getting this:

![captura de pantalla 2015-01-14 a la s 14 23 22](https://cloud.githubusercontent.com/assets/2271201/5739414/9d910fbe-9bf9-11e4-8419-97432c0e55a8.png)



![captura de pantalla 2015-01-14 a la s 14 23 33](https://cloud.githubusercontent.com/assets/2271201/5739415/9f8d8c48-9bf9-11e4-9452-2490d2bc0810.png)

![pic5](https://cloud.githubusercontent.com/assets/2271201/5739445/e4d6e6a0-9bf9-11e4-92dd-b92acf1396bd.png)


Note, the tagespiel icon isnt even showed.

Any hint or what can I do to fix this issue ?

Thanks !

**Comment by jkphl** [comment]
Could you please send me the original SVG files and your config object to joschi@kuphal.net. I'll try to figure out what's the problem.

**Comment by Avcajaraville** [comment]
Already sent you that email ! 

Thanks again for your help !

**Comment by jkphl** [comment]
Thanks! Will try to work on it later today / tonight! :)

**Comment by Avcajaraville** [comment]
Great, that would be awesome !

I cant wait to use svg all around my site, but till I have the sprites fixed, I cant deploy them.

Thanks so much for helping :)

**Comment by jkphl** [comment]
Ok, it turned out this is a major bug in *svg-sprite*! I know the reason, now I'm working on a fix. Please give me some more ...

**Comment by jkphl** [comment]
As I said, this was a major bug in *svg-sprite* which got fixed with [release v1.0.11](https://github.com/jkphl/svg-sprite/tree/v1.0.11).

However, you've still got a problem with your original SVG files: Several of them are using the same CSS class names (inside a `<defs>` block), e.g. `st0` or `st1`. They are generated by Illustrator when exporting the SVG files and interfere with each other when they get compiled into the sprite ("CSS bleeding"). *svg-sprite* doesn't namespace CSS class names at the moment (in contrast to IDs), so in addition to the above fix you will also need to sanitize your SVG files (manually, I guess).

Please let me know if this works for you! I'll close this issue for now, but please feel free to re-open in case of further problems.

Cheers, Joschi

**Comment by Avcajaraville** [comment]
Amazing ! That was the problem...

Anyway, the whole svg collection I have is huge, and some svgs have at least 60 st. classes, which means, sanitize it manually is kind of not an option.

So, how people address this problem ? Alternatives to Illustrator that doesn't make this class naming ? Any plugin, or something I can use to randomize the naming of theses classes and ids ?

BTW; was awesome that you had a look to my problem and found the solution. Really. Thank you so much ! We have been struggling with this for ages :D

**Comment by jkphl** [comment]
No problem, glad it worked for you!

In fact, I'm thinking about implementing class namespacing into *svg-sprite* (as I did it for IDs), as this seems to be an essential problem under certain circumstances. I just never thought of it. Like with ID namespacing, however, this will has to be an optional feature as it can break e.g. scripting / animating the SVGs. I created a new issue for that (#42).

Cheers, Joschi
